app/src/main/java/io/github/trojan_gfw/igniter/MainActivity.java

```diff
@@ -342,6 +351,44 @@ public void onRun() {
         }
     }
 
+    @Override
+    protected void onResume() {
+        super.onResume();
+
+        // use handler to getPrimaryClip
+        mHandler.post(new Runnable() {
+            @Override
+            public void run() {
+                ClipboardManager mClipboardManager = (ClipboardManager) getSystemService(CLIPBOARD_SERVICE);
+                // check clipboard
+                if (!mClipboardManager.hasPrimaryClip() || mClipboardManager.getPrimaryClip().getItemCount() == 0) {
+                    return;
+                }
+
+                final CharSequence clipboardText = mClipboardManager.getPrimaryClip().getItemAt(0).getText();
+                // check scheme
+                if (!"trojan".equals(Uri.parse(clipboardText.toString()).getScheme())) {
+                    return;
+                }
+                // show once if trojan url
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                    mClipboardManager.clearPrimaryClip();
+                }
+                new AlertDialog.Builder(MainActivity.this)
+                        .setMessage(R.string.clipboard_import_tip)
+                        .setPositiveButton(R.string.common_confirm, new DialogInterface.OnClickListener() {
+                            @Override
+                            public void onClick(DialogInterface dialog, int which) {
+                                trojanURLText.setText(clipboardText);
+                            }
+                        })
+                        .setNegativeButton(R.string.common_cancel, null)
+                        .create()
+                        .show();
+            }
+        });
+    }
```

Comment:
Hi Cha0sX, if you just want to use `Handler` for inserting the action at the rear of events queue, I suggest calling `Threads.instance().runOnUiThread(Runnable)` instead.

Reply:
Done👌